resources/js/components/TheEntriesList.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <v-container
+  class="my-0 mx-auto py-0"
+  style="max-width: 800px;"
+  >
+
+    <template v-for="e in groupedEntries">
+      <v-row
+        v-if="e.type === 'sub'"
+        class="mt-8"
+        :key="e.sub + e.date"
+      >
+        <v-col
+          class="pl-4 text-uppercase grey--text text--darken-2"
+          style="font-size: .875rem; font-weight: 600;"
+        >
+          {{ e.date }}
+        </v-col>
+        <v-col class="d-none d-md-flex" />
+        <v-col
+          v-html="formatValue(e.value)"
+          class="text-uppercase grey--text text--darken-2 text-right pr-8"
+          style="font-weight: 600;"
+        />
+      </v-row>
+      <an-entry-card
+        v-else
+        class="mb-4"
+        :value="e"
+        :key="e.id"
+      />
+    </template>
+  </v-container>
+</template>
+
+<script>
+  import { differenceInCalendarDays, format } from 'date-fns'
+  import AnEntryCard from '@/js/components/AnEntryCard'
+  export default {
+    name: 'TheEntriesList',
+    components: {
+      AnEntryCard,
+    },
+    props: {
+      entries: {
+        type: Array,
+        default: () => [],
+      },
+    },
+    computed: {
+      groupedEntries () {
+        let day, daydex, diff
+        return this.entries.reduce((grp, e, i) => {
+          // are we on a new date?
+          const date = format(new Date(e.date), 'y-MM-dd')
+          if (date !== day) {
+            // yes, create a new date subheader
+            day = date
+            daydex = grp.length
+            diff = differenceInCalendarDays(new Date(), new Date(e.date))
+            const label = diff === 0 ? 'Today' : diff === 1 ? 'Yesterday' : format(new Date(e.date), 'EEE, d MMM')
+            grp.push({
+              type: 'sub',
+              date: label,
+              value: parseFloat(e.value),
+            })
+          } else {
+            // no, update the total for the current date
+            grp[daydex].value += parseFloat(e.value)
+          }
+          grp.push({ type: 'entry', ...e })
+          return grp
+        }, [])
+      },
```

Comment:
Please could you explain your thought process here? Is there a reason you opted to group the data on the front-end?

Reply:
Thank you for the feedback!

The simple answer, I've been doing so much SPA work lately, it was my first instinct to do it on the client.

That said, if dates for each entry are stored in UTC then what counts as "today" is relative to the geographical location of the client. If an entry happens on 20 Jul 2020 at 11PM EDT, the ISO 8601 date would be `2020-07-21T03:00:00.000Z`.  It's the same point in time but the representations have two different dates, which affects grouping.

Also, my app only groups one page of results, i.e. the `visibleEntries`, at a time. Now that I think about it, I did not consider the likely case when a group of entries breaks across a page boundary. That would definitely result in the last group total on the page being wrong.

More generally, I've taken the approach of getting ALL of the entries from the API at page load, and only interacting with the API when there are changes on the frontend. It would allow the entries to be stored/cached on the client. Depending on responsiveness, use case, and security concerns, I can just as easily see doing the manipulations on the server side, e.g. with an aggregrate query or the like.